web/satellite/src/views/Register.vue

```diff
@@ -105,81 +169,84 @@ import { createUserRequest } from '@/api/users';
 
                 if (this.$data.repeatedPassword !== this.$data.password) {
                     this.$data.repeatedPasswordError = 'Password doesn\'t match';
-                    hasError = true;
+                    return false;
                 }
 
                 if (!this.$data.isTermsAccepted) {
                     this.$data.isTermsAcceptedError = true;
-                    hasError = true;
+                    return false;
                 }
 
-                if (hasError) return;
-
+                return true;
+            },
+            createUser: async function() {
                 let user = {
-                    email,
-                    firstName,
-                    lastName,
+                    email: this.$data.email.trim(),
+                    firstName: this.$data.firstName.trim(),
+                    lastName: this.$data.lastName.trim(),
+                    secret: this.$data.secret.trim(),
                 };
 
                 let response = await createUserRequest(user, this.$data.password);
+
                 if (!response.isSuccess) {
                     this.$store.dispatch(NOTIFICATION_ACTIONS.ERROR, response.errorMessage);
+                    this.$data.loadingClassName = LOADING_CLASSES.LOADING_OVERLAY;
 
                     return;
                 }
 
                 this.$store.dispatch(APP_STATE_ACTIONS.TOGGLE_SUCCESSFUL_REGISTRATION_POPUP);
-            }
-        },
-        data: function (): RegisterData {
+            },
+            onCreateClick: function () {
+                let self = this as any;
 
-            return {
-                firstName: '',
-                firstNameError: '',
-                lastName: '',
-                email: '',
-                emailError: '',
-                password: '',
-                passwordError: '',
-                repeatedPassword: '',
-                repeatedPasswordError: '',
-                isTermsAccepted: false,
-                isTermsAcceptedError: false,
-            };
+                if (!self.validateFields()) return;
+
+                this.$data.loadingClassName = LOADING_CLASSES.LOADING_OVERLAY_ACTIVE;
+
+                self.createUser();
+            },
+            onLogoClick: function (): void {
+                location.reload();
+            },
+            onLoginClick: function (): void {
+                this.$router.push(ROUTES.LOGIN.path);
+            },
+        },
+        computed: {
+            infoImage: function() {
+                return EMPTY_STATE_IMAGES.INFO
+            },
         },
-        computed: {},
         components: {
-            HeaderedInput,
-            Checkbox,
-            Button,
+            HeaderlessInput,
             RegistrationSuccessPopup
+        },
+        mounted(): void {
+            let token: any = this.$route.query.token ? this.$route.query.token.toString() : null;
+
+            if(token) {
+                this.$data.secret = token;
+            }
```

Comment:
if(this.$route.query.token) {
                this.$data.secret = this.$route.query.token.toString();
            }